### ISSUE TITLE
Unable to log in with self-hosted gitlab

### ISSUE THREAD

**Issue by me-cs** [codecov/self-hosted#34]
I'm trying to deploy self-hosted codecov and use it in conjunction with self-hosted gitlab
I configured the gitlab application. 
as shown below
![image](https://github.com/codecov/self-hosted/assets/49519858/803effa5-a853-4a77-9fd8-a4ddcf8a6068)
After deploying codecov using docker compose, I was successfully prompted to login with gitlab ce
![image](https://github.com/codecov/self-hosted/assets/49519858/7bad2c2d-89e2-4953-b1e0-533285481a38)
And then this error appeared
![image](https://github.com/codecov/self-hosted/assets/49519858/a4a3ee48-512e-4220-baba-b29b09e9c9ae)
The difference I found was that I configured the callback address to be 
http://ip:port/login/gitlab_enterprise
but it changed to
http://ip:port/login/gle
![image](https://github.com/codecov/self-hosted/assets/49519858/ec0d3619-360d-4be3-995c-31ea1cb188a2)
When I change it to gitlab_enterprise and type it directly into the browser, I can bring up the gitlab authentication window, but it still fails after authentication
![image](https://github.com/codecov/self-hosted/assets/49519858/614455ec-6ec4-428f-a0d2-c6b455c5922a)

Any help is appreciated!

**Comment by ChrisZhangJin** [comment]
we encounted the same issue, and didn't find any useful information.

**Comment by fmdoe** [comment]
Does it work with ```https://codecov-selfhosted-hostname.com/login/gle``` as redirect URI in Gitlab?  

![gle](https://github.com/codecov/self-hosted/assets/117662748/737f139a-32c6-4db9-a989-9cffd47b387f)

(replace gitlab_enterprise with gle)

**Comment by ChrisZhangJin** [comment]
> Does it work with `https://codecov-selfhosted-hostname.com/login/gle` as redirect URI in Gitlab?
> 
> ![gle](https://private-user-images.githubusercontent.com/117662748/318760140-737f139a-32c6-4db9-a989-9cffd47b387f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMTIyMTksIm5iZiI6MTcxMjExMTkxOSwicGF0aCI6Ii8xMTc2NjI3NDgvMzE4NzYwMTQwLTczN2YxMzlhLTMyYzYtNGRiOS1hOTg5LTljZmZkNDdiMzg3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwM1QwMjM4MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNjVkMTEwYWNjODE5M2Y0MzE4NWFiMWQxODMwZjY4NTg1YmExYWJmYjY3ZGViZmJlZTZlMjQ2MzllYTdkNzE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2nqYkrRPgtYist1C10ezws4-wUG5Ww_1MlV7BVbCnlY)
> 
> (replace gitlab_enterprise with gle)

no, actually, i'd tried both way of login/gle and login/gitlab_enterprise. Neither works.

From the worker's log, it showed 401 which means the authenication is failed i guess when it jumped back from gitlab to codecov.

i'm using the Community version of gitlab, so not sure if it is kind of incompatiblity.

**Comment by hverlin** [comment]
Seeing the same issue. Even after fixing the redirect URL and using `gle`, it does not work.
http://localhost:8080/internal/user fails with 
```
{"detail":"Authentication credentials were not provided."}
```

**Comment by tekkeitserktok** [comment]
Hello,

Any updates on this bug ? Facing it too...

Thanks !

**Comment by bert0RM** [comment]
This fixed the issue for me: https://github.com/codecov/self-hosted/pull/42
Can you please test this on your end?

**Comment by tekkeitserktok** [comment]
Hello @bert0RM, it doesn't work but as I said in another comment, I use MY_IP.nip.io for codecov and gitlab.local domains for my local setup.
